README.md

```diff
@@ -2,140 +2,167 @@
+```python
+my_cluster_folder = rh.folder(path="./my_folder").to(gpu, path="~/my_folder")
+my_s3_folder = my_cluster_folder.to("s3", path="my_bucket/my_folder")
+my_local_folder = my_s3_folder.to("here")
+```
 
-Here is **all the code you need** to stand up a stable diffusion inference service on
-a fresh cloud GPU.
+You can send a function to the cluster, including the environment in which the function will live, which is actually
+set up in its own Ray process. You can send it to an existing env, or create a new one on the fly. Like the folder 
+above, the function object which is returned from `.to` is a proxy to the remote function. When you call it a 
+lightweight request is sent to the cluster's Runhouse HTTP server to execute the function with the given inputs and 
+returns the results. Note that the function is not serialized, but rather imported on the cluster after the local 
+working directory (`"./"`, by default the git root) is sent up.
 
 
 ```python
-import runhouse as rh
 from diffusers import StableDiffusionPipeline
 
 def sd_generate(prompt):
     model = StableDiffusionPipeline.from_pretrained("stabilityai/stable-diffusion-2-base").to("cuda")
-    return model(prompt).images[0]
-
-gpu = rh.ondemand_cluster(name="my-a100", instance_type="A100:1").up_if_not()
-sd_generate = rh.function(sd_generate).to(gpu, env=["torch", "diffusers"])
+    return model(prompt).images
 
-# the following runs on our remote A100 gpu
-sd_generate("An oil painting of Keanu Reeves eating a sandwich.").show()
-sd_generate.save(name="sd_generate")
+if __name__ == "__main__":
+    remote_sd_generate = rh.function(sd_generate).to(gpu, env=["./", "torch", "diffusers"])
+    imgs = remote_sd_generate("A hot dog made out of matcha.")
+    imgs[0].show()
 ```
 
-On the data side, sync folders, tables, or blobs between local, clusters, and file storage. All
-this is done without bouncing off the laptop.
+The above env list is a shorthand, but you can create quite elaborate envs (even quite a bit more elaborate than the 
+example below). Local folders are sent (rsynced) up as needed, and the environment setup is cached so it only reruns 
+if something changes.
 
 ```python
-import runhouse as rh
-
-folder_on_gpu = rh.folder(path="./instance_images").to(gpu, path="dreambooth/instance_images")
-
-folder_on_s3 = folder_on_gpu.to("s3", path="dreambooth/instance_images")
-folder_on_s3.save("dreambooth_outputs")
+my_env = rh.env(name="my_env",
+                reqs=["torch", "diffusers", "~/code/my_other_repo"],
+                setup_cmds=["source activate ~/.bash_rc"],
+                env_vars={"TOKEN": "1234"},
+                workdir="./")  # Current git root
+my_env = my_env.to(gpu)  # Called implicitly if passed to Function.to, like above
 ```
 
-Reuse your saved compute and data resources from anywhere, with a single line of Python.
+rh.Function is actually a special case of rh.Module, which is how we send classes to clusters. There are a number of
```

Comment:
nit, everywhere you refernece Python code I'd add monospace code delimiters, e.g. `rh.Function`, `rh.Module`